### ISSUE TITLE
Intersection type not resolved correctly for array callable

### ISSUE THREAD

**Issue by derrabus** [vimeo/psalm#8941]
Following code example:

```php
interface A {}

interface B
{
    public function doSomething(mixed $value): void;
}

class MyClass {
    public function __construct(
        private readonly A&B $ab,
    ) {}
    
    public function doSomething(array $values): void
    {
        array_walk($values, [$this->ab, 'doSomething']);
    }
}
```

https://psalm.dev/r/c0232e8d70

Psalm complains:

> ERROR: [UndefinedMethod](https://psalm.dev/022) - 17:29 - Method A::dosomething does not exist

I believe that this is a false positive. `doSomething()` is a method of `B`, so it must be present on the intersection `A&B`.

**Comment by derrabus** [comment]
This regression must've been introduced between Psalm 5.2.0 and 5.3.0.
